cdk/lib/clamav-scanner-stack.ts

```diff
@@ -0,0 +1,75 @@
+import { Fn, Stack,
+         aws_ecr as ecr,
+         aws_ecs as ecs,
+         aws_iam as iam,
+         aws_ssm as ssm,
+         aws_s3 as s3,
+         aws_s3_notifications as s3n,
+         aws_lambda_nodejs as lambdaNodejs
+       } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { ClamavScannerStackProps } from './clamav-scanner-stack-props';
+import { parseEcrAccountId, parseEcrRegion } from './common-stack-funcs';
+import { ClamavScan } from './clamav-scan';
+
+export class ClamavScannerStack extends Stack {
+
+  readonly lambda: lambdaNodejs.NodejsFunction;
+
+  constructor(scope: Construct, id: string, props: ClamavScannerStackProps) {
+    super(scope, id, props);
+
+    const clamavRepo = ecr.Repository.fromRepositoryArn(this, 'clamavRepo', `arn:aws:ecr:${parseEcrRegion(props.envProps.REGISTRY)}:${parseEcrAccountId(props.envProps.REGISTRY)}:repository/${props.envProps.REPOSITORY}/clamav`);
+
+    const clamavTaskDef = new ecs.FargateTaskDefinition(this, 'clamavTaskDef', {
+      cpu: props.clamavTaskDef.taskCpu,
+      memoryLimitMiB: props.clamavTaskDef.taskMem,
+    });
+
+    // Allow clamavScan to manage S3 files
+    const pCkanCloudstorageContainerName = ssm.StringParameter.fromStringParameterAttributes(this, 'pCkanCloudstorageContainerName', {
+      parameterName: `/${props.environment}/opendata/ckan/cloudstorage_container_name`,
+    });
+    const bucketArn = `arn:aws:s3:::${pCkanCloudstorageContainerName.stringValue}`;
+
```

Comment:
Instead of passing the bucket name through parameter store, I would just add it to stack props, its not really a secret. Passing non-secrets through parameter store is just overly complex, even though this parameter already is there from previous implementations.